egs/swbd/s5c/local/reverberate/reverberate_data_dir.py

```diff
@@ -0,0 +1,165 @@
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+def Corrupt_wav_noise(input_dir, output_dir, rir_list, noise_list, snr_array, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    noises = list_cyclic_iterator(return_nonempty_lines(open(noise_list).readlines()), random_seed = 1)
+    snrs = list_cyclic_iterator(snr_array)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            noise = noises.next().split()
+            noise_file_location = noise[-1]
+            snr = snrs.next()
+            command_list.append("{4} {0} wav-reverberate --noise-file={2} --snr-db={3} - {1} - |\n".format(wav_pipe, rir_file_location, noise_file_location, snr, output_wav_id))
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+
+def Replicate_file1(input_file, output_file, num_replica, prefix):
+    list = map(lambda x: x.strip(), open(input_file))
+    f = open(output_file, "w")
+    for i in range(num_replica):
+        for line in list:
+            split1 = line.split()
+            split1[0] = prefix + str(i) + "_" + split1[0]
+            print(" ".join(split1), file=f)
+    f.close()
+
+
+def Replicate_file2(input_file, output_file, num_replica, prefix):
+    list = map(lambda x: x.strip(), open(input_file))
+    f = open(output_file, "w")
+    for i in range(num_replica):
+        for line in list:
+            split1 = line.split()
+            split1[0] = prefix + str(i) + "_" + split1[0]
+            split1[1] = prefix + str(i) + "_" + split1[1]
+            print(" ".join(split1), file=f)
+    f.close()
+
+
+def Main():
+    args = GetArgs()
+
+    snr_array = [20 , 15, 10 , 5, 0]
+    if args.noise_list is not None:
+        Corrupt_wav_noise(args.input_dir, args.output_dir, args.rir_list, args.noise_list, snr_array, args.num_replica, "rvb")
+    else:
+        Corrupt_wav(args.input_dir, args.output_dir, args.rir_list, args.num_replica, "rvb")
+    Replicate_file2(args.input_dir + "/utt2spk", args.output_dir + "/utt2spk", args.num_replica, "rvb")
+    train_lib.RunKaldiCommand("utils/utt2spk_to_spk2utt.pl <{output_dir}/utt2spk >{output_dir}/spk2utt"
+                    .format(output_dir = args.output_dir))
+
+    if os.path.isfile(args.input_dir + "/text"):
+        Replicate_file1(args.input_dir + "/text", args.output_dir + "/text", args.num_replica, "rvb")
```

Comment:
Choosing better names for the methods Replicate* would make the code more readable.